### DISCUSSION TITLE
Some basic starting questions

### DISCUSSION THREAD

**Discussion by jcedens** [AxisCommunications/discussions#29] [ACAP] [answered]
Hey all, I've run into some roadblocks while trying to start using Docker tools and work through the tutorials.

I've installed Docker on my Windows 10 machine along with GIT as recommended in the documentation. My expectation is that I will use the Docker environment to create applications that will be run on some Axis cameras I manage. One application I'm particularly interested in would set a virtual output to 1 when the room is bright and to 0 when the room is dark. This will be sensed by my VMS to trigger continuous recording when the room lights are on. Because this is an older camera with firmware that only goes to 9.80.3.13, I would expect that I would be using SDK 3.1.

I created an account on docker.com - jcedens - and attempted to use the Docker Pull Command "docker pull axisecp/acap-sdk".  The responses were:

Using default tag: latest
Error response from daemon: manifest for axisecp/acap-sdk:latest not found: manifest unknown: manifest unknown

Do I need to download acap3-sdk directly from github and somehow import it into Docker locally?

Another question I have is - does the Docker file itself contain the Ubuntu image? The reason I ask that is because when I run "wsl --list --all" in a Command Prompt, the responses are:

Windows Subsystem for Linux Distributions:
docker-desktop (Default)
docker-desktop-data

Do I need to install Ubuntu 22.04 on my computer before acap-sdk will work?

Thanks...

**Comment by pataxis** [comment, marked as answer]
Hi @jcedens ,
There is no `latest` tag in the `acap-sdk` image repo and first release was [3.2](https://help.axis.com/acap-3-developer-guide#whats-new-in-release-3-2-0).

The 3.1 release was distributed as a Debian package (.deb), i.e. possible to install on a Debian or Ubuntu distribution. Contact the [Axis partner program](https://www.axis.com/partner/technology-integration-partner-program) for access to these SDKs.

For the 3.1 release there were also two helper Docker images called `acap-toolchain` and `acap-api`, see [3.1 release notes](https://help.axis.com/acap-3-developer-guide#whats-new-in-release-3-1-0), that are now used in the background when we build the `acap-sdk` image. Note that these helper images are only available for `aarch64` and `armv7hf`, not the two architectures `mips` (Artpec-5) and  `armv5tejl` (HiSilicon chips) that stopped on 9.80.

**Reply by jcedens** [reply]
Hi @pataxis, thanks for the quick and detailed reply.

If I understand you correctly, it sounds like the 3.1 release tools are not available from Docker Hub or GitHub, only from the Partner Program - is that correct?

The cameras I am looking and doing this with use armv7hf architecture, so that should be okay.

I looked over the application for the Partner Program and it looks like it makes a requirement to provide a contact who is "authorized to sign the Technology Integration Partner Program agreement." I work for a University and have no idea who would be authorized to sign such a document, but I'm sure that I'm not. Would such an official agreement be required in order for the Partner Program to provide the SDK?

Thanks again,

**Reply by pataxis** [reply]
The Debian packages are only available from the Partner Program yes. Regarding the agreement and the SDK, I'm not the right person to answer but I will see if I can find someone that can.

The Docker Hub helper images are of course publicly available. I will look in to if I can make something more helpful in this regard and get back to you.

**Reply by jcedens** [reply]
@pataxis, thanks very much for that - I went ahead and used the agreement input form to explain the situation and referenced that you had suggested that I contact them. Appreciate your help!

**Reply by pataxis** [reply]
Hi again, we decided to create the `acap-sdk` image for architectures `aarch64` and `armv7hf` of version 3.1.
You find the release artifacts on:

- Docker Hub [acap-sdk 3.1](https://hub.docker.com/r/axisecp/acap-sdk/tags?page=1&name=3.1)
- GitHub [acap3-examples tag 3.1](https://github.com/AxisCommunications/acap3-examples/tree/3.1)

**Reply by jcedens** [reply]
@pataxis , thanks - you guys are great! This kind of support is why I insist that our campus only buys Axis cameras.

**Comment by mms-** [comment]
All this seems overly convoluted. Will armv7hf cameras currently stuck on 9.8 firmware ever be updated to support SDK >= 3.2 so we can focus on development and not funny stuff with partner programs and agreements?)

**Reply by pataxis** [reply]
Hi @mms- , we made an extra release 3.1 of image `acap-sdk` for `armv7hf` and `aarch64` which maps to Axis OS 9.80. Please open the full thread of the selected answer for more information.
